updates on the using-the-API page on gen3.org (#124)
uc-cdis/gen3.org: content/resources/user/using-api.md

>__Note:__ The Gen3 commons uses GraphQL as the language for querying metadata across Gen3 Data Commons. To learn the basics of writing queries in GraphQL, please visit: <http://graphql.org/learn>.

Gen3 features a variety of API endpoints such as `/submission`, `/index`, or `/graphql`, which differ in how they access the resource and contain each a subset of REST (Representational State Transfer) APIs for networked applications. REST APIs are restricted in their interactions via HTTP request methods such as GET, POST, PATCH, PUT, or DELETE. The GET request retrieves data in read-only mode, POST sends data and creates a new resource, PATCH updates/modifies a resource, PUT updates/replaces a resource, and DELETE deletes a resource. At Gen3, the GET endpoint `/api/v0/submission/<program>/<project>/_dictionary` will for example get the dictionary schema for entities of a project, or `/api/v0/submission/graphql/getschema` will get the data dictionary schema in .JSON format.

> Further API specifications of the Gen3 mirocroservices can be browsed in [the developer documentation](/resources/developer/microservice) and (REST) API requests documentation of a Gen3 microservice can be found on swagger (e.g. [Sheepdog](https://petstore.swagger.io/?url=https://raw.githubusercontent.com/uc-cdis/sheepdog/master/openapi/swagger.yml#/), [Peregrine](https://petstore.swagger.io/?url=https://raw.githubusercontent.com/uc-cdis/peregrine/master/openapis/swagger.yaml)).

Reviewer: ```suggestion
Further API specifications of the Gen3 microservices can be browsed in [the developer documentation](/resources/developer/microservice) and the (REST) API Swagger documentation for each Gen3 microservice can be found in the microservice's documentation on GitHub (e.g. [Sheepdog](https://petstore.swagger.io/?url=https://raw.githubusercontent.com/uc-cdis/sheepdog/master/openapi/swagger.yml#/), [Peregrine](https://petstore.swagger.io/?url=https://raw.githubusercontent.com/uc-cdis/peregrine/master/openapis/swagger.yaml)).
```

print(dl.text) # display response

# Data Upload via API Endpoint Request:
tsv_file = '/cmd-test.tsv' # loading the tsv file that contains text string and tab-separated values
tsv = ''
count = 0
with open(tsv_file, 'r') as file:
    for line in file:
        if count == 0:
            header = line
        tsv = tsv + line + "\r"
        count += 1
program_name,project_code = <program_name>, <project_code>
u = requests.put('https://gen3.datacommons.io/api/v0/submission/{}/{}'.format(program_name,project_code), data=tsv, headers=headers)
u.text # should display the API response

```

> If an an error such as "You don't have access... " occurs, then the API key is most likely out of date and a new access token will need to be made. Further errors could occur if the uploaded file is not correctly formatted for the Gen3 data model.

Reviewer: ```suggestion
If an an error such as "You don't have access... " occurs, then either you do not have access, or the API key is out of date and a new access token will need to be made. Further errors could occur if the uploaded file is not correctly formatted for the Gen3 data model.
```
